src/inline/plugin.py

```diff
@@ -362,193 +365,202 @@ def parse_constructor(self, node):
+                            raise MalformedException(f"inline test: {self.arg_repeated_str} must be greater than 0")
+                        self.cur_inline_test.repeated = value
+                    elif arg_idx == ConstrArgs.TAG_STR:
+                        tags = []
+                        
+                        if sys.version_info < (3, 8, 0):
+                            elt_type = ast.Str
+                        else:
+                            elt_type = ast.Constant
+                        
+                        for elt in arg.elts:
+                            value = getattr(elt, value_prop_name)
+                            if (not isinstance(elt, elt_type) and isinstance(value, str)):
+                                raise MalformedException(f"tag can only be List of string")
+                            tags.append(value)
+                        self.cur_inline_test.tag = tags
+                    # For non-special cases, set the attribute defined by the dictionary
+                    else:
+                        setattr(self.cur_inline_test,
+                                property_names[arg_idx],
+                                getattr(arg, value_prop_name))
+                    
+                   
+                    ## Match implementation of above conditional tree; commented since Python < 3.10 does not support match
+                    
+                    # match arg_idx:
+                    #     case ConstrArgs.REPEATED:
+                    #         if arg.value <= 0:
+                    #             raise MalformedException(f"inline test: {self.arg_repeated_str} must be greater than 0")
+                    #         self.cur_inline_test.repeated = getattr(arg, value_prop_name)
+                    #     case ConstrArgs.TAG_STR:
+                    #         tags = []
+                    #         for elt in arg.elts:
+                    #             if not (isinstance(elt, ast.Constant) and isinstance(elt.value, str)):
+                    #                 raise MalformedException(f"tag can only be List of string")
+                    #             tags.append(getattr(elt, value_prop_name))
+                    #         self.cur_inline_test.tag = tags
+                    #     # For non-special cases, set the attribute defined by the dictionary
+                    #     case _:
+                    #         setattr(self.cur_inline_test,
+                    #                 property_names[arg_idx],
+                    #                 getattr(arg, value_prop_name))
+                else:
+                    raise MalformedException(
+                        f"inline test: {self.class_name_str}() accepts {NUM_OF_ARGUMENTS} arguments. 'test_name' must be a string constant, 'parameterized' must be a boolean constant, 'repeated' must be a positive integer, 'tag' must be a list of string, 'timeout' must be a positive float"
+                    )
+              
```

Comment:
Cool, it looks so much cleaner now.

We may keep the lowest supported Python version to be lower than 3.10 for a while, so cannot completely switch to `match` for now...